site/src/pages/DeploySettingsPage/ServiceBannerSettingsPage.tsx

```diff
@@ -0,0 +1,245 @@
+import makeStyles from "@material-ui/core/styles/makeStyles"
+import FormControlLabel from "@material-ui/core/FormControlLabel"
+import Switch from "@material-ui/core/Switch"
+import { BlockPicker } from "react-color"
+import { useTheme } from "@material-ui/core/styles"
+import FormHelperText from "@material-ui/core/FormHelperText"
+import Button from "@material-ui/core/Button"
+
+export const Language = {
+  messageLabel: "Message",
+  backgroundColorLabel: "Background Color",
+  updateBanner: "Update",
+  previewBanner: "Preview",
+}
+
+export interface ServiceBannerFormValues {
+  message?: string
+  backgroundColor?: string
+  enabled: boolean
+}
+
+// TODO:
+const validationSchema = Yup.object({})
+
+// ServiceBanner is unlike the other Deployment Settings pages because it
+// implements a form, whereas the others are read-only. We make this
+// exception because the Service Banner is visual, and configuring it from
+// the command line would be a significantly worse user experience.
+const ServiceBannerSettingsPage: React.FC = () => {
+  const xServices = useContext(XServiceContext)
+  const [serviceBannerState, serviceBannerSend] = useActor(
+    xServices.serviceBannerXService,
+  )
+
+  const [entitlementsState] = useActor(xServices.entitlementsXService)
+
+  const serviceBanner = serviceBannerState.context.serviceBanner
+
+  const styles = useStyles()
+
+  const isEntitled =
+    entitlementsState.context.entitlements.features[FeatureNames.ServiceBanners]
+      .entitlement !== "not_entitled"
+
+  const setBanner = (values: ServiceBannerFormValues, preview: boolean) => {
+    const newBanner = {
+      message: values.message,
+      enabled: values.enabled,
+      background_color: values.backgroundColor,
+    }
+    if (preview) {
+      serviceBannerSend({
+        type: "SET_PREVIEW_BANNER",
+        serviceBanner: newBanner,
+      })
+      return
+    }
+    serviceBannerSend({
+      type: "SET_BANNER",
+      serviceBanner: newBanner,
+    })
+  }
+
+  const initialValues: ServiceBannerFormValues = {
+    message: serviceBanner.message,
+    enabled: serviceBanner.enabled,
+    backgroundColor: serviceBanner.background_color,
+  }
+
+  const form: FormikContextType<ServiceBannerFormValues> =
+    useFormik<ServiceBannerFormValues>({
+      initialValues,
+      validationSchema,
+      onSubmit: (values) => setBanner(values, false),
+    })
+  const getFieldHelpers = getFormHelpers<ServiceBannerFormValues>(form)
+
+  const [backgroundColor, setBackgroundColor] = useState(
+    form.values.backgroundColor,
+  )
+
+  const theme = useTheme()
+
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle("Service Banner Settings")}</title>
+      </Helmet>
+
+      <Header
+        title="Service Banner"
+        description="Configure a banner that displays a message to all users"
+        docsHref="https://coder.com/docs/coder-oss/latest/admin/high-availability#service-banners"
```

Comment:
Is this the right link? I didn't see anything about service banners on the HA page.